README.mingw.md

```diff
@@ -1,58 +1,180 @@
+macOS.
+
+You can build for 32bit or 64bit Windows. If your intention is to use a native
+Uniscribe usp10.dll from Windows 7 or before, you would need to build for 32bit.
+If you want to use a native DirectWrite DLL from Windows 10 or later, you would
+need to build for 64bit.
 
 1. Install Wine.
-   - Fedora: `dnf install wine`.
 
-2. Install `mingw-w64` compiler.
-   - Fedora, 32bit: `dnf install mingw32-gcc-c++`
-   - Fedora, 64bit: `dnf install mingw64-gcc-c++`
-   - Debian: `apt install g++-mingw-w64`
-   - Mac: `brew install mingw-w64`
+  - Fedora: `dnf install wine`.
+  - Mac: `brew install wine-stable`.
+
+2. Install the `mingw-w64` cross-compiler.
+
+  - Fedora, 32bit: `dnf install mingw32-gcc-c++`
+  - Fedora, 64bit: `dnf install mingw64-gcc-c++`
+  - Debian: `apt install g++-mingw-w64`
+  - Mac: `brew install mingw-w64`
+
+3. Install dependencies.
+
+  - Fedora, 32bit: `dnf install mingw32-glib2 mingw32-cairo mingw32-freetype`
+  - Fedora, 64bit: `dnf install mingw64-glib2 mingw64-cairo mingw64-freetype`
+
+Or you could simply pull them all in one go with:
+
+  - Fedora, 32bit: `dnf install mingw32-harfbuzz`
+  - Fedora, 64bit: `dnf install mingw64-harfbuzz`
+
+If you cannot find these packages for your distribution, or you are on macOS,
+you can skip to the next step, as meson will automatically download and build
+the dependencies for you.
+
+4. If you are familiar with `meson`, you can use the cross-compile files we
+provide to find your way around. Read until the end of this section before
+deciding which one to use.
+
+  - 32bit: `meson --cross-file=.ci/win32-cross-file.txt build-win -Dglib-enabled -Dcairo=enabled -Dgdi=enabled -Ddirectwrite=enabled`
+  - 64bit: `meson --cross-file=.ci/win64-cross-file.txt build-win -Dglib-enabled -Dcairo=enabled -Dgdi=enabled -Ddirectwrite=enabled`
+
+In which case, you will proceed to run `ninja` as usual to build:
+
+  - `ninja -C build-win`
+
+Or you can simply invoke the scripts we provide for our Continuous Integration
+system, to configure and build HarfBuzz for you. This is the easiest way to
+build HarfBuzz for Windows and how we build our Windows binaries:
+
+  - 32bit: `./.ci/build-win.sh 32 && ln -s build-win32 build-win`
+  - 64bit: `./.ci/build-win.sh 64 && ln -s build-win64 build-win`
+
+This might take a while, since, if you do not have the dependencies installed,
+meson will download and build them for you.
+
+5. If everything succeeds, you should have the `hb-shape.exe`, `hb-view.exe`,
+`hb-subset.exe`, and `hb-info.exe` executables in `build-win/util`.
+
+6. Configure your wine to find system mingw libraries. While there, set it also
+to find the built HarfBuzz DLLs:
+
+  - Fedora, 32bit: `export WINEPATH="$HOME/harfbuzz/build-win/src;/usr/i686-w64-mingw32/sys-root/mingw/bin"`
+  - Fedora, 64bit: `export WINEPATH="$HOME/harfbuzz/build-win/src;/usr/x86_64-w64-mingw32/sys-root/mingw/bin"`
+  - Other systems: `export WINEPATH="$HOME/harfbuzz/build-win/src"`
```

Comment:
This might not be needed if one uses `meson devenv` to run the commands, since it sets `WINEPATH`.

Reply:
With the `exe_wrapper` manual change, yes?